Sources/Kevin/Core/Custom/KevinWebView.swift

```diff
@@ -49,27 +49,27 @@ internal class KevinWebView: WKWebView, WKNavigationDelegate {
             decisionHandler(.allow)
             return
         }
-
-        if shouldProcessExternally(url: url) {
-            if #available(iOS 10.0, *) {
-                UIApplication.shared.open(url, options: [.universalLinksOnly: true]) { isSuccess in
-                    if isSuccess {
-                        decisionHandler(.cancel)
-                    } else {
-                        decisionHandler(.allow)
-                    }
-                }
-            } else {
-                decisionHandler(.allow)
-            }
-        } else {
-            if isCallbackURL(url: url) {
+        
+        func finalize(url: URL, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+            if url.absoluteString.starts(with: callbackURL.absoluteString) {
                 decisionHandler(.cancel)
                 completedCallback?(url, nil)
             } else {
                 decisionHandler(.allow)
             }
         }
+
+        if #available(iOS 10.0, *), shouldProcessExternally(url: url) {
+            UIApplication.shared.open(url, options: [.universalLinksOnly: true]) { isSuccess in
+                if isSuccess {
+                    decisionHandler(.cancel)
+                } else {
+                    finalize(url: url, decisionHandler: decisionHandler)
```

Comment:
Do we really need to `finalize` if URL is `shouldProcessExternally == true`? 
1. Our callback should never be processed externally anyway
2. Additionally, if callback URL is setup as a deeplink for app that is using SDK - it might fall in to `shouldProcessExternally` in an attempt to open itself.
Example:
- App ABC uses `https://ABC.callback.url` as an associated domain to open an app via deeplink
- Same url (`https://ABC.callback.url`) is used as a callback
- In theory `shouldProcessExternally` would return true and there will be an attempt to open ABC app instead of processing it as a callback. If this attempt will be successful or not is up to OS implementation which is outside of our control.

Reply:
We do that to check if it failed to open the universal link. That was the issue. If the universal link is opened, we just stay there until the user has handled the app and returned.

Reply:
Are we sure that `callbackUrl` will never be successfully opened (falling into `decisionHandler(.cancel)` and never processing because it never hit the `finalize`)?